### DISCUSSION TITLE
Special interface for PDF-file in CefSharp

### DISCUSSION THREAD

**Discussion by yura2015** [cefsharp/CefSharp#4370] [General]
When a PDF file is uploaded to CefSharp, a special interface is visible in the window. Can you tell me in which CefSharp class this interface is edited (created)? I need to change (hide some elements, namely "more actions" and "rotatecounterclockwise").
v107.1.4

And is there an opportunity and how to change (hide some elements, namely "more actions" and "rotatecounterclockwise") the interface of such a file opened in CefSharp on the page page using JavaScript, for example: pdfObj.data = path + '#page=' + page + '&zoom=1&view=FitH&scrollbar=0'; ? How it worked in IE 6.0-8.0.

**Comment by amaitland** [comment]
> Can you tell me in which CefSharp class this interface is edited (created)?

The `PDF Viewer(PDFium)`  is implemented entirely in `Chromium`. There's no options for customising these features currently. You could possibly disable the `PDF Viewer` and use say https://mozilla.github.io/pdf.js/ to render the `PDF`. I've not tried this in recent version.

You can try posting on https://www.magpcss.org/ceforum/viewforum.php?f=7 to see if support can be added.

> for example: pdfObj.data = path + '#page=' + page + '&zoom=1&view=FitH&scrollbar=0';

Some of those arguments should work if passed as part of the `URL`. Basically it's whatever the underlying version of `Chromium` supports.

Try open the `URL` directly to see if they work.

**Reply by yura2015** [reply]
Thanks for the help. And could you clarify in which Chromium class this is implemented, (maybe in which files, if you know)? This class(s) is there in the source texts that are available in this project? I would really like to understand how it works.

**Reply by amaitland** [reply]
From memory the PDF Viewer UI is actually implemented in HTML, it's probably in one of the `pak` files. 

You can search the Chromium Source online at https://cs.chromium.org/

**Reply by yura2015** [reply]
I have sources from https://github.com/cefsharp/CefSharp /. Are they not the same ones? I was trying to find a description of the interface for PDF files right here.
If the interface description is in pak files, then of course I can't do anything about it..
Disable the PDF viewer and use https://mozilla.github.io/pdf .js is unrealistic for me. I can't imagine how this is possible at all.
Thanks for the information.

**Reply by amaitland** [reply]
> I have sources from https://github.com/cefsharp/CefSharp /. Are they not the same ones? I was trying to find a description of the interface for PDF files right here.

You won't find the `PDF Viewer` in this repository, it's implemented in [Chomium](https://cs.chromium.org/).

Read the first few paragraphs of https://github.com/cefsharp/CefSharp/wiki/General-Usage to get a better understanding of the project.

> Disable the PDF viewer and use https://mozilla.github.io/pdf .js is unrealistic for me. I can't imagine how this is possible at all.

It's likely easier to use `pdf.js` than compile your own custom patched chromium.

Otherwise you might need to look into another non `CEF` based solution.

**Reply by yura2015** [reply]
Good. Thanks for the information.
